base/inference.jl

```diff
@@ -3245,11 +3257,12 @@ end
 
 #### finalize and record the result of running type inference ####
 
-function isinlineable(m::Method, src::CodeInfo)
+function isinlineable(m::Method, src::CodeInfo, mod::Module, params::InferenceParams, bonus::Int=0)
     # compute the cost (size) of inlining this code
     inlineable = false
-    cost = 1000
+    cost_threshold = params.inline_cost_threshold
     if m.module === _topmod(m.module)
+        # a few functions get special treatment
```

Comment:
I'm not entirely clear on how one would do that automatically. It's also not a very big performance hit. I think I made those changes when this was at an earlier stage and it caused a bigger performance hit. In particular [this line](https://github.com/JuliaLang/julia/blob/abcdc77b58a6f58f9cf376e820aefb7e445ee7fd/base/inference.jl#L4666), which  runs for any `:call` expr that's not an Intrinsic/Builtin which means that it's a dynamic dispatch---if it were statically known it would be an `:invoke`---is probably what mostly fixes it.